code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/AuthDao.java

```diff
@@ -60,7 +60,7 @@
 
     Account getAccountByLogin(String publicValue, String secretValue, TransformationService transformationService);
 
-    String getRole(Account account, Policy policy);
+    String getRole(Account account, Policy policy, Policy authenticatedAsPolicy);
```

Comment:
As the name suggests authenticatedAsPolicy - is the policy object created for the account logged in to Rancher

The policy object is for the account resource for which the request is applicable - in ase of create stack request, it will be the policy for the environment/project - account of kind 'project'